src/Service/Storage/Ef/EfGlobalGlobalStorage.cs

```diff
@@ -51,6 +52,13 @@ public async Task<int> UpdatePeriodicCredentialReportsAsync()
         return rows;
     }
 
+    public Task<ApiKeyDesc> GetApiKeyAsync(string apiKey)
+    {
+        var appId = ApiKeyUtils.GetAppId(apiKey);
+        var pk = apiKey.Substring(apiKey.Length - 4);
+        return _db.ApiKeys.FirstOrDefaultAsync(e => e.Id == pk && e.Tenant == appId);
+    }
+
```

Comment:
question: While OK, just curious, why was this moved?

Reply:
When validating the api keys and api secrets, there is no valid authorization. Claims are not populated, so we cannot use the EfTenantStorage. If we choose to keep the EfTenantFactory, we could set the appId manually with ManualTenantProvider. And then we could keep it on EfTenantStorage.

Reply:
Yeah I think I'd prefer this code to be part of the TenantStorage, and use the factory to get a TenantStorage while validating them. Just makes it less of a risk that we accidentally call this method.

Reply:
Ok moved.